osu.Game/Beatmaps/IBeatmap.cs

```diff
@@ -53,6 +53,10 @@ public interface IBeatmap
         /// </summary>
         IReadOnlyList<HitObject> HitObjects { get; }
 
+        int[] Bookmarks { get; set; }
+
+        CountdownType Countdown { get; set; }
```

Comment:
So while I don't disagree with doing this in the first place (since I'm the one who suggested it), I don't think this goes far enough. At the *least* the same thing should also be done to `CountdownOffset`:

https://github.com/ppy/osu/blob/b4a64701f5baa923d7ce77a692f271a0e55a16a3/osu.Game/Beatmaps/BeatmapInfo.cs#L179-L182

because it's weird that now half of the countdown info is in `IBeatmap` and half of it is in `BeatmapInfo`. But really, the same logic can be extended to [everything in this `#region`](https://github.com/ppy/osu/blob/b4a64701f5baa923d7ce77a692f271a0e55a16a3/osu.Game/Beatmaps/BeatmapInfo.cs#L137-L184).

If there is a time to move that stuff out of there, now is probably one of the best times to do so. @ppy/team-client would appreciate opinions on this matter.

Reply:
I was afraid to remove it since I didn't want to cause any unnecessary Realm changes. Albeit, rethinking it now, I don't think it would cause many issues. I'll wait for a response from others however (and I'll probably do another pull request just for this if this is the direction we want to go ahead with).

I'm a bit curious however about how I would name the new class if we decide to go in this direction, I'm currently thinking of `BeatmapDetails` but I'm unsure if that'll stick.

Reply:
#23432 strangely relevant here.

I agree with moving all of these out as a first step.

> If we do decide to move everything then it may be better to wrap all of that in some object rather than move everything to `IBeatmap`, because `ProgressiveCalculationBeatmap` is going to grow by quite a large margin if moving properties 1:1. But maybe that's fine.

I think if anything it should be a separate interface, rather than a separate class.

Reply:
> I think if anything it should be a separate interface, rather than a separate class.

I decided to make a proper interface and a class called `IBeatmapSettings` so that it can be easily added to `IBeatmap`, and it condenses all of the options into one proper area without inflating the size of classes such as `ProgressiveCalculationBeatmap` with properties.